m4/acx_crypto_backend.m4

```diff
@@ -65,9 +65,9 @@ AC_DEFUN([ACX_CRYPTO_BACKEND],[
 		AC_MSG_RESULT(OpenSSL)
 
 		if test "x${enable_fips}" = "xyes"; then
-			ACX_OPENSSL(1,0,1)
+			ACX_OPENSSL(1,0,2)
 		else
-			ACX_OPENSSL(1,0,0)
+			ACX_OPENSSL(1,0,2)
```

Comment:
If-branch and else-branch are the same.

Reply:
And advice on version for FIPS compliance?

Reply:
According to [FIPS-140](https://www.openssl.org/docs/fips.html) openssl version should be 1.0.2, but you are changing openssl version in both branches to 1.0.2. Indeed for FIPS it should stay at version 1.0.2.

It's odd that two branches uses the same version. May be we should add a note that version of OpenSSL for FIPS compatibility should stay exactly at version 1.0.2?

But I guess there is an issue with openssl version checking for FIPS. In acx_openssl.m4 an openssl version should be [at least](https://github.com/opendnssec/SoftHSMv2/blame/9f63ccc0529894b546a2b18c03197f7e30aab6ae/m4/acx_openssl.m4#L43) as specified version in ACX_OPENSSL. [OpenSSL FIPS notes](https://www.openssl.org/docs/fips.html) that "Neither validation will work with any release other than 1.0.2".

Reply:
So guess it makes sense to keep it as-is, and the else branch will increase later as needed?

Reply:
Yep. And it makes sense to add a comment why it has else branch and when it should be increased.

Reply:
Done